src/CAN/can_mapping.c

```diff
@@ -47,7 +49,30 @@ float canmapping_extract_value(uint64_t raw_data, const CANMapping *mapping)
                         break;
                 }
         }
-        return (float)raw_value;
+
+        /* convert type */
+        float extracted_value = 0;
+        switch (mapping->type) {
+        	case CANMappingType_IEEE754:
+        		extracted_value = *((float*)&raw_value);
+        		break;
+        	case CANMappingType_signed:
+        		if (mapping->length <= 8) {
+        			extracted_value = *((int8_t*)&raw_value);
```

Comment:
Umm... So I am super confused about what is happening here.  Raw value is an 8 bit float pointer?  Why are you using a `uint32_t` to represent an address that may not even be 32 bits in length?  That is very unportable.  Should probably be a `void*` since that will adjust with varying address sizes.  And even if you use that the conversion here seems super off.  You are stuffing a signed int like value into a float!?!  Something ain't stirring the Koolaid well here.